### PR TITLE
Fix backend test runs inside and outside Docker

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- Followup to #3418 

## Changes Proposed

- Readd the `testDbStart` and `testDbStop` tasks to `build.gradle`, but make them conditional on a new `skipDbSetup` property. Thus, when running the backend tests outside of docker, the db setup is handled for your. But when running inside docker with the `yarn test:backend` command, the db setup will be skipped by gradle since it is already handled by the yarn script.
- Add some caveats to the README regarding the current lack of seed data
- Change usages of `docker-compose` to `docker compose`

## Additional Information

- decisions that were made
- discussion of tradeoffs / future work
- complaints about how bad the code is

### Infrastructure
- [ ] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes (including new error messages) have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/build.gradle [hunk 1]
```diff
@@ -194,6 +194,20 @@ spotless {
     }
 }
 
+task testDbStart(type: Exec) {
+    environment "SR_DB_PORT", testDbPort
+    commandLine "docker-compose", "-p", "simple-report-tests", "-f", "../docker-compose.yml", "up", "-d", "db"
```
Comment (zdeveloper) [inline backend/build.gradle:199]: this needs to be `docker compose` (with space)

**Review by zdeveloper — DISMISSED** [review]
LGTM! thank you for the updates!

File: backend/build.gradle (hunk 1 above)
Comment (nickclyde) [reply backend/build.gradle:199]: Good call! I'll try to fix other spots that still have the dash too

**Review by emmastephenson — DISMISSED** [review]
LGTM once Zedd's comments are addressed!

**Review by emmastephenson — APPROVED** [review]

**Review by zdeveloper — APPROVED** [review]
Thank you!

**Pull request merged**